OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -550,6 +550,16 @@ void CloseWindow()
 					return true;
 			};
 
+			chatTextField.OnEscKey = () =>
+			{
+				if (chatTextField.Text.Length == 0)
+					closeLobby();
```

Comment:
We deliberately chose not to do this because leaving the lobby is a high-impact action for something that can be so easily triggered.